### PR TITLE
Read environment variables

### PR DESCRIPTION
**Pull request by mooreryan**
Addresses https://github.com/gleam-lang/cookbook/issues/15.

- I based the format on the [working with files](https://github.com/gleam-lang/cookbook/blob/e3cbe86d15e31746083ea431f6b5cc5e0d5e49c2/universal/test/file_system/work_with_files.gleam) test.
- I used the category of "operating system" as that is what the [Rust cookbook](https://rust-lang-nursery.github.io/rust-cookbook/#operating-system) uses for reading environment variables.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,3 +32,7 @@ Examples showing how to do many things in Gleam!
 - [Rendering HTML](./universal/test/formats/rendering_html.gleam)
 - [Rendering JSON](./universal/test/formats/rendering_json.gleam)
 - [Rendering XML](./universal/test/formats/rendering_xml.gleam)
+
+## Operating system
+
+- [Read environment variable](./universal/test/operating_system/read_environment_variable.gleam)
```
Comment (lpil) [inline README.md]:
```suggestion
- [Read environment variables](./universal/test/operating_system/read_environment_variables.gleam)
```

File: universal/test/operating_system/read_environment_variable.gleam [hunk 2]
```diff
@@ -0,0 +1,40 @@
+//// # Read environment variables
+//// 
+//// The envoy package is used to work with environment variables.  This package 
+//// works on both Erlang and JavaScript targets.
+//// 
+//// Note that we are using `let assert` to crash if any of these functions 
+//// fail.  In a real application or library, you would want to handle results 
+//// properly.
```
Comment (lpil) [inline universal/test/operating_system/read_environment_variable.gleam]: Move this into the test to match the others please.

File: universal/test/operating_system/read_environment_variable.gleam [hunk 3]
```diff
@@ -0,0 +1,40 @@
+//// # Read environment variables
+//// 
+//// The envoy package is used to work with environment variables.  This package
```
Comment (lpil) [inline universal/test/operating_system/read_environment_variable.gleam]:
```suggestion
//// The envoy package is used to work with environment variables. This package 
```

File: universal/test/operating_system/read_environment_variable.gleam [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+//// works on both Erlang and JavaScript targets.
+//// 
+//// Note that we are using `let assert` to crash if any of these functions 
+//// fail.  In a real application or library, you would want to handle results 
+//// properly.
+//// 
+//// ## Dependencies
+//// 
+//// - https://hex.pm/packages/envoy
+////
```
Comment (lpil) [inline universal/test/operating_system/read_environment_variable.gleam]:
```suggestion
```

File: universal/test/operating_system/read_environment_variable.gleam [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+//// 
+
+import envoy
+import gleam/dict
+
+pub fn main_test() {
+  // First, unset the PORT variable.  
+  //
+  // We do this fist to get the environment into a known state.  If the PORT
+  // variable is not already set, this will have no effect.
```
Comment (lpil) [inline universal/test/operating_system/read_environment_variable.gleam]: The comments should describe how perform each task, it shouldn't be telling a story. See the other examples for the style

File: universal/test/operating_system/read_environment_variable.gleam [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  let Nil = envoy.unset("PORT")
```
Comment (lpil) [inline universal/test/operating_system/read_environment_variable.gleam]:
```suggestion
  envoy.unset("PORT")
```

File: universal/test/operating_system/read_environment_variable.gleam [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  // Try to get an environment variable by name.  
+  //
+  // We expect that this will fail, as we unset this variable above.
+  let assert Error(Nil) = envoy.get("PORT")
+
+  // Set the PORT environment variable to 8080.
+  let Nil = envoy.set("PORT", "8080")
```
Comment (lpil) [inline universal/test/operating_system/read_environment_variable.gleam]:
```suggestion
  envoy.set("PORT", "8080")
```

**Review by lpil — COMMENTED** [review]
Thank you! I've left some notes

File: universal/test/operating_system/read_environment_variable.gleam (hunk 3 above)
Comment (mooreryan) [reply universal/test/operating_system/read_environment_variable.gleam]: All double spaces after period in prose has been converted to a single space.

File: universal/test/operating_system/read_environment_variable.gleam (hunk 5 above)
Comment (mooreryan) [reply universal/test/operating_system/read_environment_variable.gleam]: Hopefully it now better matches what you want.  It's more similar to the work_with_files.gleam example now.

**Review by lpil — APPROVED** [review]
Beautiful! Thank you for waiting too, CodeBEAM conf took over my life for a few weeks

**Comment by lpil** [comment]
Oops, the tests are failing!

**Comment by mooreryan** [comment]
> Oops, the tests are failing!

Looks like the javascript and erlang behavior on key not found is different.  See here: https://github.com/lpil/envoy/issues/3

**Review by lpil — APPROVED** [review]
Ahhh good find! Thank you so much!!

**Pull request merged**